addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURLUnitTest.java

```diff
@@ -0,0 +1,427 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrules;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import net.htmlparser.jericho.Source;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Test;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.extension.httpsessions.HttpSessionsParam;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+public class TestInfoSessionIdURLUnitTest extends PassiveScannerTest<TestInfoSessionIdURL> {
+
+    private HttpMessage msg;
+    private static final String BODY = "Some text in the response, doesn't matter.\nLine 2\n";
+
+    @Override
+    protected TestInfoSessionIdURL createScanner() {
+
+        TestInfoSessionIdURL scanner = new TestInfoSessionIdURL();
+        //        scanner.setConfig(new ZapXmlConfiguration());
+        setUpHttpSessionsParam();
+        return scanner;
+    }
+
+    protected HttpMessage createHttpMessageWithRespBody(String responseBody)
+            throws HttpMalformedHeaderException, URIException {
+
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI("http://example.com", false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+        msg.setResponseBody(responseBody);
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n"
+                        + "Server: Apache-Coyote/1.1\r\n"
+                        + "Content-Type: text/plain\r\n"
+                        + "Content-Length: "
+                        + responseBody.length()
+                        + "\r\n");
+        return msg;
+    }
+
+    @Test
+    public void shouldHaveSessionIdsInConfig() throws Exception {
+
+        // Given
+        OptionsParam options = Model.getSingleton().getOptionsParam();
+        HttpSessionsParam sessionOptions = options.getParamSet(HttpSessionsParam.class);
+
+        // When
+        List<String> sessionIds = new ArrayList<String>();
+        if (sessionOptions != null) {
+            sessionIds = sessionOptions.getDefaultTokensEnabled();
+        }
+
+        // Then
+        assertThat(sessionIds, is(not(empty())));
+    }
+
+    @Test
+    public void containsJSESSIONIDAsUrlParameter()
```

Comment:
This should be generic (e.g. `constainsSessionIdAsUrlParameter`) the scanner is not checking just `JSESSIONID`.